### PR TITLE
chore: Update gradle version to 7.2

### PR DESCRIPTION
**Pull request by vaish3496**
#### References to other issues or PRs
Fixes: #20 

#### Brief description of what is fixed or changed
Updated the grade version from 5.0 to 6.8.3 to support the latest OpenJDK version.

ss1

### CODE REVIEW

**Comment by ravisuhag** [comment]
@vaish3496 This issue was created some time back. We can actually upgrade Gradle to the latest version.

**Comment by vaish3496** [comment]
> @vaish3496 This issue was created some time back. We can actually upgrade Gradle to the latest version.

Are you suggesting v7.2? I'm not sure if that is stable. Or do you want me to upgrade it to v7.1.1?

**Comment by ravisuhag** [comment]
@vaish3496 Yes, we can go to 7.2.

**Comment by vaish3496** [comment]
Getting this error while trying to upgrade Gradle to 7.2. Not sure how to deal with this.

Capture

**Comment by ravisuhag** [comment]
@vaish3496 You would need look at all breaking APIs for gradle , migration guide etc. to make build.gradle compatible to 7.2

File: gradlew [hunk 1]
```diff
@@ -28,16 +28,16 @@ APP_NAME="Gradle"
 APP_BASE_NAME=`basename "$0"`
 
 # Add default JVM options here. You can also use JAVA_OPTS and GRADLE_OPTS to pass JVM options to this script.
-DEFAULT_JVM_OPTS=""
+DEFAULT_JVM_OPTS='"-Xmx64m"'
```
Comment (lavkesh) [inline gradlew:31]: IMO 64Mb is not enough heap.

File: gradlew [hunk 2]
```diff
@@ -28,16 +44,16 @@ APP_NAME="Gradle"
 APP_BASE_NAME=`basename "$0"`
 
 # Add default JVM options here. You can also use JAVA_OPTS and GRADLE_OPTS to pass JVM options to this script.
-DEFAULT_JVM_OPTS=""
+DEFAULT_JVM_OPTS='"-Xmx64m" "-Xms64m"'
```
[the remark this answers is not in the document]
Comment (vaish3496) [reply gradlew:47]: This is changed by default while updating the Gradle. I can set it to an empty string if that's the case.

**Comment by sumitaich1998** [comment]
change protobuf-gradle-plugin version to 0.8.17
`        classpath 'com.google.protobuf:protobuf-gradle-plugin:0.8.8'`

on line 6 in `build.gradle`
[](https://github.com/vaish3496/firehose/blob/29691086934593c4fe0310d0b6bcf812ad6cd1c9/build.gradle#L6)

**Comment by ravisuhag** [comment]
@vaish3496 Build is failing with deprecated feature of gradle still being used in build file. We would need to upgrade them all. Check build logs for the same.

**Comment by sumitaich1998** [comment]
can we go up to Gradle 8.0 ? @vaish3496

**Comment by vaish3496** [comment]
> @vaish3496 Build is failing with deprecated feature of gradle still being used in build file. We would need to upgrade them all. Check build logs for the same.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0. But works well with 7.2.

**Comment by vaish3496** [comment]
> can we go up to Gradle 8.0 ? @vaish3496

8.0 is yet to be released. 7.2 is the latest.

**Comment by ravisuhag** [comment]
@vaish3496  https://github.com/odpf/firehose/pull/101/checks?check_run_id=3469295039#step:5:29

**Comment by ravisuhag** [comment]
@sumitaich1998 @vaish3496 Some tests are failing. Can you folks check and fix those.

File: gradlew.bat [hunk 3]
```diff
@@ -13,15 +29,18 @@ if "%DIRNAME%" == "" set DIRNAME=.
 set APP_BASE_NAME=%~n0
 set APP_HOME=%DIRNAME%
 
+@rem Resolve any "." and ".." in APP_HOME to make it shorter.
+for %%i in ("%APP_HOME%") do set APP_HOME=%%~fi
+
 @rem Add default JVM options here. You can also use JAVA_OPTS and GRADLE_OPTS to pass JVM options to this script.
-set DEFAULT_JVM_OPTS=
+set DEFAULT_JVM_OPTS="-Xmx64m" "-Xms64m"
```
Comment (lavkesh) [inline gradlew.bat:36]: could you remove this also ?

Comment (lavkesh) [reply gradlew.bat:36]: this is for windows.

**Comment by vaish3496** [comment]
@ravisuhag @sumitaich1998 @lavkesh all checks have been passed. Thanks!

**Comment by lavkesh** [comment]
👍🏻 lgtm

**Comment by ravisuhag** [comment]
@vaish3496 Thanks for the contirbutions. Looking forward to more :)

**Pull request merged**